sn_node/src/bin/safenode/main.rs

```diff
@@ -296,6 +296,25 @@ fn monitor_node_events(mut node_events_rx: NodeEventsReceiver, ctrl_tx: mpsc::Se
     let _handle = tokio::spawn(async move {
         loop {
             match node_events_rx.recv().await {
+                Ok(NodeEvent::AttemptingNetworkConnection) => {
+                    Marker::AttemptingNetworkConnection.log();
+                    trace!("Attempting to connect to the network: dialing initial peers");
+                    let inactivity_timeout = Instant::now();
+
+                    loop {
+                        if inactivity_timeout.elapsed() >= Duration::from_secs(60) {
+                            Marker::NetworkConnectionTimingOut.log();
+                            trace!("Attempt to dial initial peers and connect to the network is timing out after {:?}", inactivity_timeout);
+                        }
+
+                        match node_events_rx.recv().await {
+                            Ok(NodeEvent::ConnectedToNetwork) => {
```

Comment:
If other events are coming in here, we might not be handling them down in the real event handlers. E.g. `NodeEvent::BehindNat`, etc.

In case of `ConnectedToNetwork` this will prevent `Marker::NodeConnectedToNetwork.log()` from being called by that handler.

Reply:
Added an is_connected boolean to handle this instead.